microsoft/testsuites/security/fips.py

```diff
@@ -26,20 +30,60 @@ class Fips(TestSuite):
         priority=3,
     )
     def verify_fips_enable(self, log: Logger, node: Node) -> None:
-        result = node.execute("command -v fips-mode-setup", shell=True)
-        if result.exit_code != 0:
-            raise SkippedException(
-                "Command not found: fips-mode-setup. "
-                f"Please ensure {node.os.name} supports fips mode."
-            )
+        if isinstance(node.os, CBLMariner):
+            # check If It is a non-FIPS image
+            if node.execute("rpm -qa | grep dracut-fips").exit_code != 0:
+                raise SkippedException(
+                    "Not a fips enabled image."
+                )
+            else:
+                # FIPS image
+                result = node.tools[Cat].run(
+                    "/proc/sys/crypto/fips_enabled", sudo=True, force_run=True)
+
+                if result.exit_code != 0:
+                    raise LisaException(
+                        "fips_enabled file is not found in proc file system. "
+                    )
+
+                if "1" != result.stdout:
+                    raise LisaException(
+                        "fips is not enabled properly. "
+                        f"Please ensure {node.os.name} has fips turned "
+                        "on by default."
+                    )
+
+                result = node.execute("openssl md5")
+                print(result)
+                # md5 should not work If It is a FIPS image
+                # Following the output of the above command
+                # Error setting digest
+                # 131590634539840:error:060800C8:digital envelope routines:EVP_DigestInit_ex:disabled for FIPS:crypto/evp/digest.c:135:
+                if result.exit_code != 0:
+                    pattern = re.compile(r"Error setting digest\n(.+?disabled for FIPS:crypto.+?)\n")
+                    if not get_matched_str(result.stdout, pattern):
+                        raise LisaException(
+                            "openssl is not operating under fips mode."
```

Comment:
`is not operating` is expected or not? please give the clear exception message

Reply:
I have added more detailed messages. Please take a look at the latest commit.